### PR TITLE
Direct Beta/Dev/Nightly downloads on old OS versions to ESR (Issue #13317)

### PR DESCRIPTION
**Pull request by alexgibson**
## One-line summary

Firefox desktop is ending support on OS versions less than or equal to Windows 8.1, and the same on macOS 10.14 and below. Existing users on these systems are being migrated to Firefox ESR.

Because of this, there is also an ask to prevent people downloading Firefox on these systems in the future from the website. They should instead be directed toward Firefox ESR. This needs to happen before Firefox 116 is released (August 1st).

Firefox 116 is soon to hit Beta however (July 4th), so we need to tackle pre-release channels first as a priority.

That's what this PR sets out to do by:

- Hiding Beta/Dev Edition/Nightly download buttons on <= Win 8.1 and <= macOS 10.14.
- Displaying a message in its place directing people to download Firefox ESR instead.

## Significant changes and points to review

This needs some careful testing. I'd suggest using BrowserStack. Please test using both IE and the latest versions of other browsers on these old operating systems.

## Issue / Bugzilla link

#13317

## Screenshots

image

## Testing

- http://localhost:8000/en-US/firefox/channel/desktop/
- http://localhost:8000/en-US/firefox/developer/

Demo:

- https://www-demo2.allizom.org/en-US/firefox/channel/desktop/
- https://www-demo2.allizom.org/en-US/firefox/developer/

On Win 8.1 and below:

- [x] Firefox ESR 64/32bit download links are shown instead of regular Beta/Dev/Nightly links on both of the above pages.
- [x] Clicking each ESR link gives the correct 32/64bit installer.

On Windows 10 and above:

- [x] The regular Beta/Dev/Nightly links are displayed as normal on both pages.

On macOS 10.14 and below:

- [x] A Firefox ESR download link is shown instead of regular Beta/Dev/Nightly links on both of the above pages.
- [ ] Clicking the ESR link gives the correct macOS installer.

On macOS 10.15 and above:

- [x] The regular Beta/Dev/Nightly links are displayed as normal on both pages.

### CODE REVIEW

**Comment by alexgibson** [comment]
Successful test run: https://github.com/mozilla/bedrock/actions/runs/5422705415

**Comment by alexgibson** [comment]
Note for reviewer: there's been zero time for L10n here, but initially folks will just have to get this message in English :/

File: bedrock/firefox/templates/firefox/includes/download-button.html [hunk 1]
```diff
@@ -50,6 +50,46 @@
...
+    {% elif channel == 'alpha' %}
+      {% set channel_name = 'Firefox Developer Edition' %}
+    {% elif channel == 'nightly' %}
+      {% set channel_name = 'Firefox Nightly' %}
+    {% else %}
+      {% set channel_name = 'Firefox' %}
+    {% endif %}
+
+    <div class="fx-unsupported-message win" data-nosnippet="true">
+      <p><strong>{{ ftl('download-button-unsupported-platform', channel_name=channel_name, help_url='https://support.mozilla.org/kb/firefox-users-macos-1012-1013-1014-moving-to-extended-support', os_version='Windows 8.1') }}</strong></p>
```
Comment (stevejalim) [inline bedrock/firefox/templates/firefox/includes/download-button.html:65]: Why is this a mac link but os_version references Windows 8.1?

File: bedrock/firefox/templates/firefox/includes/download-button.html [hunk 2]
```diff
@@ -50,6 +50,46 @@
...
+        <li>
+          <a class="mzp-c-button mzp-t-product download-link" href="https://download.mozilla.org/?product=firefox-latest-ssl&os=win&lang={{ LANG }}" data-link-type="download" data-download-os="Desktop" data-download-version="win" data-display-name="Firefox Extended Support Release">
+            {{ ftl('download-firefox-esr-32') }}
+          </a>
+        </li>
+      </ul>
+    </div>
+
+    <div class="fx-unsupported-message mac" data-nosnippet="true">
+      <p><strong>{{ ftl('download-button-unsupported-platform', channel_name=channel_name, help_url='https://support.mozilla.org/kb/firefox-users-windows-7-8-and-81-moving-extended-support', os_version='macOS 10.14') }}</strong></p>
```
Comment (stevejalim) [inline bedrock/firefox/templates/firefox/includes/download-button.html:82]: Similarly, the link here is for Windows but the os_version refs macOS

File: bedrock/firefox/templates/firefox/includes/download-button.html (hunk 1 above)
Comment (alexgibson) [reply bedrock/firefox/templates/firefox/includes/download-button.html:65]: This message is for Windows users

Comment (stevejalim) [reply bedrock/firefox/templates/firefox/includes/download-button.html:65]: But the help_url points to a MacOS-specific SUMO page?

Comment (alexgibson) [reply bedrock/firefox/templates/firefox/includes/download-button.html:65]: Ah, the href is the wrong way around - nice catch!

**Review by stevejalim — APPROVED** [review]
Did a drive-through on Zoom with Alex, too - looking good. 

Only fix needed is swapping the SUMO links, as noted in comments

r+wc 🔥 🦊

**Review by stevejalim — APPROVED** [review]
r+

**Pull request merged**